File: WebApi/Models/OrderModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace WebApi.Models {

	public class OrderModel {

		[Required]
		public int OrderId { get; set; }
		[Required]
		public DateTime StartDate { get; set; }
		[Required]
		public DateTime EndDate { get; set; }
		public DateTime? ActualEndDate { get; set; }
		[Required]
		public decimal Price { get; set; }
		public decimal? TotalPrice { get; set; }
		[Required]
		public int UserId { get; set; }
		[Required]
		public int VehicleId { get; set; }
	}
}

File: AngularClient/src/app/ts/classes/models/order.model.ts
import { OrderI, UserI, VehicleI } from '../../interfaces/model.interfaces';

export class Order implements OrderI {
	public orderId: number
	public startDate: Date
	public endDate: Date
	public actualEndDate: Date
	public price: number
	public totalPrice: number
	public userId: number
	public vehicleId: number
	public isActive: boolean
	constructor(
		public user: UserI,
		public vehicle: VehicleI,
	) { }
}
